### PR TITLE
Added details for #7 - Expected Technology

### PR DESCRIPTION
**Pull request by cyrussamante**
## 🔗 Related Issue(s)

- #7

## 📝 Description

Added the discussed expected technologies that would be used for the development plan report.

These are the following expected technologies that the team expects to use for the project:

* GitHub: For version control.
* GitHub Projects:  Create an organized project space to track current progress.
* LaTeX: Allows for a professional-looking report, with easy-to-use syntax and commonly used in the industry.
* Make:  Easily generates the LaTeX format into PDF for a report.
* C++: The existing codebase of TPG is written in this language.
* VS Code: Personal preference for members in this project and by the supervisor.
* MuJoCo: The main simulation environment the project aims to be compatible with.
* Clang-tidy linter: An open-source, widely used linter with an easy learning curve.
* Bash: For creating scripts.
* Python: For plotting analytical data for TPG.
* Google Test: For unit testing within the project.
* Cppcheck: For static analysis.

## ✅ How Has This Been Tested?

N/A - Compiled new pdf and verified if changes were appropriate.

### CODE REVIEW

File: docs/DevelopmentPlan/DevelopmentPlan.tex [hunk 1]
```diff
@@ -110,37 +110,54 @@ \section{Proof of Concept Demonstration Plan}
...
+  \item \textbf{GitHub:} For version control.
+  \item \textbf{GitHub Projects:}  Create an organized project space to track current progress.
+  \item \textbf{LaTeX:} Allows for a professional-looking report, with easy-to-use syntax and commonly used in the industry.
+  \item \textbf{Make:}  Easily generates the LaTeX format into PDF for a report.
+  \item \textbf{C++:} The existing codebase of TPG is written in this language.
+  \item \textbf{VS Code:} Personal preference for members in this project and by the supervisor.
+  \item \textbf{MuJoCo:} The main simulation environment the project aims to be compatible with.
+  \item \textbf{Clang-tidy:} An open-source, widely used linter with an easy learning curve.
+  \item \textbf{Bash:} For creating scripts.
+  \item \textbf{Python:} For plotting analytical data for TPG.
```
[the remark this answers is not in the document]
Comment (cyrussamante) [reply docs/DevelopmentPlan/DevelopmentPlan.tex:131]: Probably something like Docker is used? What do you think?

[the remark this answers is not in the document]
Comment (richardl2003) [reply docs/DevelopmentPlan/DevelopmentPlan.tex:131]: Let's add it here for now, we'll see if we need to end up using it

[the remark this answers is not in the document]
Comment (cyrussamante) [reply docs/DevelopmentPlan/DevelopmentPlan.tex:131]: @markacruz @calvynsiong @edwarddgao, what do you think about this?

[the remark this answers is not in the document]
Comment (calvynsiong) [reply docs/DevelopmentPlan/DevelopmentPlan.tex:131]: Yea I think Docker makes sense in the future, but we can add that in later right?

File: docs/DevelopmentPlan/DevelopmentPlan.tex [hunk 2]
```diff
@@ -110,37 +110,54 @@ \section{Proof of Concept Demonstration Plan}
...
+  \item \textbf{Google Test:} For unit testing within the project.
```
[the remark this answers is not in the document]
Comment (cyrussamante) [reply docs/DevelopmentPlan/DevelopmentPlan.tex:132]: I can add that sure.

File: docs/DevelopmentPlan/DevelopmentPlan.tex (hunk 1 above)
[the remark this answers is not in the document]
Comment (cyrussamante) [reply docs/DevelopmentPlan/DevelopmentPlan.tex:131]: Yup, its only a first revision. So any changes we can add / remove anytime.

[the remark this answers is not in the document]
Comment (cyrussamante) [reply docs/DevelopmentPlan/DevelopmentPlan.tex:131]: I think we can add this in later.

File: docs/DevelopmentPlan/DevelopmentPlan.tex [hunk 3]
```diff
@@ -110,37 +110,54 @@ \section{Proof of Concept Demonstration Plan}
...
-\item Pre-trained models
-\item Specific linter tool (if appropriate)
-\item Specific unit testing framework
-\item Investigation of code coverage measuring tools
-\item Specific plans for Continuous Integration (CI), or an explanation that CI
-  is not being done
-\item Specific performance measuring tools (like Valgrind), if
-  appropriate
-\item Tools you will likely be using?
+  \item \textbf{Git and GitHub:} For version control.
```
Comment (markacruz) [inline docs/DevelopmentPlan/DevelopmentPlan.tex:122]: nothing crazy but as a nit: we may also use GitLab for when we push code to GitLab side

File: docs/DevelopmentPlan/DevelopmentPlan.tex [hunk 4]
```diff
@@ -110,37 +110,54 @@ \section{Proof of Concept Demonstration Plan}
...
-Topics to discuss include the following:
+% \wss{What programming language or languages do you expect to use?  What external
+% libraries?  What frameworks?  What technologies.  Are there major components of
+% the implementation that you expect you will implement, despite the existence of
+% libraries that provide the required functionality.  For projects with machine
+% learning, will you use pre-trained models, or be training your own model?  }
+
+These are the following expected technologies that the team expects to use for the project:
 
 \begin{itemize}
```
Comment (markacruz) [inline docs/DevelopmentPlan/DevelopmentPlan.tex:121]: we can also add a test coverage tools in our CI pipeline like what they have [here](https://github.com/Eric-Jalal/cpp-pipeline/blob/develop/.github/workflows/build.yml#L30-L33). `gcov` seems to be the most used but you can def. look into it more.

**Review by calvynsiong — APPROVED** [review]

**Review by edwarddgao — COMMENTED** [review]
the list is good

**Pull request merged**